### ISSUE TITLE
Bug: ArrowTypeError when indexing logs directory with mixed score types within an evaluation

### ISSUE THREAD

**Issue by MattFisher** [meridianlabs-ai/inspect_scout#163]
## Description

When using `transcripts_from()` with a directory containing eval logs from different evaluations that have different score value types, an `ArrowTypeError` is raised during index creation.

### Reproduction

```python
from inspect_scout import transcripts_from, scan, ScanJob
from inspect_scout import columns as c

# Directory contains logs from multiple evaluations:
# - swe_lancer logs with float scores (0.0, 1.0)
# - core_bench logs with string scores ('I')

transcripts = (
    transcripts_from('./logs')
    .where(c.task_set == 'swe_lancer')
    .order_by(c.date, 'DESC')
    .limit(1)
)

job = ScanJob(scanners=[...], transcripts=transcripts)
result = scan(job)  # Raises ArrowTypeError
```

### Error

```
ArrowTypeError: ("Expected bytes, got a 'float' object", 'Conversion failed for column score with type object')
```

### Expected Behavior

The `where(c.task_set == 'swe_lancer')` filter should allow scanning to proceed, either by:
1. Filtering logs before index creation, or
2. Using a schema that accommodates mixed types in the `score` column (e.g., storing as string/variant)

### Actual Behavior

The index is built across ALL logs in the directory before the `where` filter is applied. When logs have incompatible score types, PyArrow fails to create a unified schema.

### Workaround

Pass only the specific log files instead of the directory:

```python
from pathlib import Path

# Filter files before passing to transcripts_from
swe_logs = list(Path('./logs').glob('*swe*lancer*.eval'))
transcripts = transcripts_from([str(f) for f in swe_logs]).order_by(c.date, 'DESC').limit(1)
```

### Environment

- inspect-scout: 0.4.6
- Python: 3.12
- OS: macOS

### Additional Context

The issue occurs because different evaluations can have different scorer implementations that return different value types:
- Numeric scorers return `float` (e.g., 0.0, 1.0)
- Categorical scorers return `str` (e.g., 'I', 'C')

This is valid per the inspect_ai API, but causes issues when inspect-scout tries to build a unified index.

**Comment by jjallaire** [comment]
Okay, thank you! That does it. Should have a fix shortly.
